core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -372,6 +375,39 @@ public Fate(T environment, FateStore<T> store, boolean runDeadResCleaner,
             break;
           }
         }
+        queueSizeHistory.clear();
+      } else {
+        // The property did not change, but should it based on available Fate threads? Maintain
+        // the last X minutes of available Fate threads. If always zero, then suggest that the
+        // MANAGER_FATE_THREADPOOL_SIZE be increased.
+        final long interval = Math.min(60, TimeUnit.MILLISECONDS
+            .toMinutes(conf.getTimeInMillis(Property.MANAGER_FATE_QUEUE_CHECK_INTERVAL)));
+        if (interval == 0) {
+          queueSizeHistory.clear();
+        } else {
+          if (queueSizeHistory.size() >= interval * 2) { // this task runs every 30s
+            boolean needMoreThreads = true;
+            for (Integer i : queueSizeHistory) {
+              if (i > 0) {
+                needMoreThreads = false;
```

Comment:
On a system that is consistently busy if sampling alot will probably catch this as non zero a few times, just catching threads that are just about to get work.  Could count the non-zero and make needMoreThreads based on seeing 95% of sample events being zero.

Reply:
Implemented in cf88581